task_processing/plugins/persistence/dynamodb_persistence.py

```diff
@@ -0,0 +1,75 @@
+import datetime
+import decimal
+
+import boto3.session as bsession
+from boto3.dynamodb.conditions import Key
+from pyrsistent import thaw
+
+from task_processing.interfaces.persistence import Persister
+
+
+class DynamoDBPersister(Persister):
+    def __init__(self, table_name, endpoint_url=None, session=None):
+        self.table_name = table_name
+        if not session:
+            session = bsession.Session()
+        self.ddb_client = session.client(
+            service_name='dynamodb',
+            endpoint_url=endpoint_url,
+        )
+        self.table = session.resource(
+            endpoint_url=endpoint_url,
+            service_name='dynamodb'
+        ).Table(table_name)
+
+    def read(self, task_id, comparison_operator='EQ'):
+        res = self.table.query(
+            KeyConditionExpression=Key('task_id').eq(task_id)
+        )
+        return [self._replace_decimals(item) for item in res['Items']]
+
+    def write(self, event):
+        return self.ddb_client.put_item(
+            TableName=self.table_name,
+            Item=self._event_to_item(event)
+        )
+
+    def _event_to_item(self, event):
+        thawed = thaw(event)
+        resp = {}
+        for k, v in thawed.items():
+            if type(v) is str:
+                resp[k] = {
+                    'S': v
```

Comment:
yeah, ignore my comment then. as for int <-> datetime, ideally it would only need to happen once in event_to_item and item_to_event but that doesn't matter much if we can't avoid dynamodb encoding scheme part.